tests/conftest.py

```diff
@@ -564,15 +563,26 @@ def spawn(args=[], display=None):
 
 
 @pytest.fixture()
-def hlwm_process(hlwm_spawner):
+def xvfb():
+    # start an Xvfb server (don't start Xephyr because
+    # Xephyr requires another runnig xserver already).
+    # also we add '-noreset' such that the server is not reset
+    # when the last client connection is closed.
+    with MultiscreenDisplay(server='Xvfb', extra_args=['-noreset']) as xserver:
+        os.environ['DISPLAY'] = xserver.display
+        yield xserver
+
+
+@pytest.fixture()
+def hlwm_process(hlwm_spawner, xvfb):
     """Set up hlwm and also check that it shuts down gently afterwards"""
-    hlwm_proc = hlwm_spawner(['--no-tag-import'])
-    kill_all_existing_windows(show_warnings=True)
+    hlwm_proc = hlwm_spawner(['--no-tag-import'], display=xvfb.display)
+    # kill_all_existing_windows(show_warnings=True)
```

Comment:
This can be removed, right?

Reply:
Right, I've removed both! Should I also remove the `kill_all_existing_windows()` function? It is not used anymore, but I can't exclude that it might be of use in the future.

Reply:
I'd remove it, yes.
If we will need it again, we have git to restore it :)

Reply:
Aaaand it's gone :)

Reply:
Thanks :D